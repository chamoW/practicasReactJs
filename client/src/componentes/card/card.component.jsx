import { IconButton, Typography } from "@mui/material";
import React from "react";
import "./card.css";

import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import FavoriteIcon from "@material-ui/icons/Favorite";
import ShareIcon from "@material-ui/icons/Share";

export const CardCustome = ({ name, email, id }) => {
  return (
    <div className="card-monster-container m-2">
      <Typography noWrap gutterBottom variant="h5" color="primary">
        {name}
      </Typography>
      <div className="card-monster-content">
        <div className="card-monster-media-text">
          <img
            alt="monster"
            className="card-monster-image"
            src={`https://robohash.org/${id}?set=set2&size=100x100`}
          />
          <p className="card-monster-text">{email}</p>
        </div>

        <p className="card-monster-text">
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Deserunt
          cumque pariatur impedit accusantium amet mollitia ad unde, qui
          praesentium excepturi architecto temporibus quasi facilis ab quos cum
          incidunt maxime dignissimos.
        </p>
      </div>

      <div disableSpacing className="card-monster-actions">
        <IconButton aria-label="add to favorites" color="primary">
          <FavoriteIcon />
        </IconButton>
        <IconButton aria-label="share" color="secondary">
          <ShareIcon />
        </IconButton>
        <IconButton aria-label="show more">
          <ExpandMoreIcon />
        </IconButton>
      </div>
    </div>
  );
};
