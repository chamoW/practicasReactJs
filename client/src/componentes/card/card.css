.card-monster-container {
    width: 200px;
    height: 275px;
    background-color: #95dada !important;
    border: 1px solid grey !important;
    border-radius: 5px;
    padding: 10px;
    cursor: pointer;
    -moz-osx-font-smoothing: grayscale;
    backface-visibility: hidden;
    transform: translateZ(0);
    transition: transform 0.25s ease-out !important;
    text-align: center;
    min-height: 100px !important;
}

.card-monster-content {
    display: flex;
    flex-direction: column;
    padding: 0 !important;
    text-align: justify;
    flex-grow: 1;
}

.card-monster-media-text {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    overflow: hidden;
    overflow-wrap: break-word;
}

.card-monster-text {
    display: -webkit-box;
    -webkit-line-clamp: 5;
    -webkit-box-orient: vertical;
    overflow: hidden;
}

.card-monster-container:hover {
    transform: scale(1.05);
}

.text-title {
    font-family: "Metal Mania", cursive;
    font-size: 72px;
}

.card-monster-image {
    width: 85px !important;
}

.card-monster-actions {
    padding: 0 !important;
    justify-content: space-evenly;
}
