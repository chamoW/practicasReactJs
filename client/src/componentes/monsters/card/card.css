.card-container {
  display: flex;
  flex-direction: column;
  background-color: #95dada !important;
  border: 1px solid grey !important;
  border-radius: 5px;
  padding: 25px;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  transform: translateZ(0);
  transition: transform 0.25s ease-out !important;
}

.card-container:hover {
    transform: scale(1.05);
}

.text-title{
  font-family: 'Metal Mania', cursive;
  font-size: 72px;
}

