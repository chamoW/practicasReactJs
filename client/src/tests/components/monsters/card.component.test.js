import React from "react";
import { shallow, mount, render } from "enzyme";
import { Card } from "../../../componentes/monsters/card/card.component";
import { it } from "date-fns/locale";

test("Card componente test", () => {
  expect(<Card />).toMatchSnapshot();
});
