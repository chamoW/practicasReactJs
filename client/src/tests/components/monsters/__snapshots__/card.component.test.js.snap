// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card componente test 1`] = `<Card />`;
